File: app/src/main/java/com/example/victo/ntiapp/MainActivity.java
package com.example.victo.ntiapp;

import android.os.Bundle;
import android.os.Handler;
import android.os.PersistableBundle;
import android.support.annotation.NonNull;
import android.support.v4.view.GestureDetectorCompat;
import java.lang.Runnable;
import android.support.v7.app.AppCompatActivity;
import android.util.Log;
import android.view.GestureDetector;
import android.view.MenuItem;
import android.view.MotionEvent;
import android.view.View;
import android.widget.Button;
import android.widget.TextView;
import android.view.View.OnClickListener;
import android.view.View.OnTouchListener;
import android.os.Message;
import java.util.Random;
import java.util.Arrays;

import java.util.concurrent.ThreadLocalRandom;

public class MainActivity extends AppCompatActivity implements GestureDetector.OnGestureListener, GestureDetector.OnDoubleTapListener{

    private TextView mTextMessage;
    private static final String TAG = "myMessage";

    private TextView snackText;
    private GestureDetectorCompat gestureDetector;

    MyDBHandler dbHandler;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        //mTextMessage = (TextView) findViewById(R.id.message);
        Log.i(TAG, "onCreate");

        this.gestureDetector = new GestureDetectorCompat(this,this);
        gestureDetector.setOnDoubleTapListener(this);

        Button randomizeButton = (Button)findViewById(R.id.randomizeButton);
        dbHandler = new MyDBHandler(this, null, null, 1);

        randomizeButton.setOnClickListener(
                new Button.OnClickListener(){
                    public void onClick(View v){
                        Object[] list = dbHandler.databaseToString().toArray(new String[dbHandler.databaseToString().size()]);
                        String[] snacks = Arrays.copyOf(list, list.length, String[].class);
                        String[] drinks = {"Coca Cola", "Coca Cola Cherry", "Coca Cola Vanilj", "Fanta", "Festis", "Mer", "Nocco", "Celcius", "Loka Flaska", "Loka Crush", "Loka burk", "Alovera", "Stor kaffe", "Varm Choklad", "Te", "Rosh?", "Snapple", "Pucko", "Pago juice", "Yoggi Yalla"};
                        final TextView snackText = (TextView)findViewById(R.id.snackText);
                        final TextView drinkText = (TextView)findViewById(R.id.drinkText);
                        Random rand = new Random();


                        Handler handler = new Handler(){
                          @Override
                            public void handleMessage(Message msg){
                              int randomNum = rand.nextInt(snacks.length);
                              snackText.setText(snacks[randomNum]);
                              int randomNum2 = rand.nextInt(drinks.length);
                              drinkText.setText(drinks[randomNum2]);
                          }
                        };

                        Runnable r = new Runnable() {
                            @Override
                            public void run() {
                                for(int i=0;i<50;i++){
                                    handler.sendEmptyMessage(0);
                                    try {
                                        Thread.sleep(50);
                                    }catch (InterruptedException ie){
                                        Log.i(TAG, "Thread interrupted");
                                    }

                                }
                            }
                        };
                        Thread myThread = new Thread(r);
                        myThread.start();

                        int randomNum = rand.nextInt(snacks.length);
                        snackText.setText(snacks[randomNum]);
                        int randomNum2 = rand.nextInt(drinks.length);
                        drinkText.setText(drinks[randomNum2]);
                    }
                }
        );
    }

    @Override
    public boolean onSingleTapConfirmed(MotionEvent e) {
        return false;
    }

    @Override
    public boolean onDoubleTap(MotionEvent e) {
        return false;
    }

    @Override
    public boolean onDoubleTapEvent(MotionEvent e) {
        return false;
    }

    @Override
    public boolean onDown(MotionEvent e) {
        return false;
    }

    @Override
    public void onShowPress(MotionEvent e) {

    }

    @Override
    public boolean onSingleTapUp(MotionEvent e) {
        return false;
    }

    @Override
    public boolean onScroll(MotionEvent e1, MotionEvent e2, float distanceX, float distanceY) {
        return false;
    }

    @Override
    public void onLongPress(MotionEvent e) {

    }

    @Override
    public boolean onFling(MotionEvent e1, MotionEvent e2, float velocityX, float velocityY) {
        return false;
    }

    @Override
    protected void onStart() {
        super.onStart();
        Log.i(TAG, "onStart");
    }

    @Override
    protected void onResume() {
        super.onResume();
        Log.i(TAG, "onResume");
    }

    @Override
    protected void onPause() {
        super.onPause();
        Log.i(TAG, "onPause");
    }

    @Override
    public void onSaveInstanceState(Bundle outState, PersistableBundle outPersistentState) {
        super.onSaveInstanceState(outState, outPersistentState);
        Log.i(TAG, "onInstanceState");
    }

    @Override
    protected void onStop() {
        super.onStop();
        Log.i(TAG, "onStop");
    }


}

File: app/src/main/java/com/chooser/nti/ntichooser/MainActivity.kt
package com.chooser.nti.ntichooser

import android.annotation.SuppressLint
import android.content.Intent
import android.os.Build
import android.os.Bundle
import android.os.Handler
import android.os.Message
import android.provider.AlarmClock.EXTRA_MESSAGE
import android.support.annotation.RequiresApi
import android.support.design.widget.Snackbar
import android.support.design.widget.NavigationView
import android.support.v4.view.GravityCompat
import android.support.v7.app.ActionBarDrawerToggle
import android.support.v7.app.AppCompatActivity
import android.util.Log
import android.view.Menu
import android.view.MenuItem
import android.view.View
import android.widget.Button
import android.widget.EditText
import android.widget.Switch
import android.widget.TextView
import kotlinx.android.synthetic.main.activity_main.*
import kotlinx.android.synthetic.main.app_bar_main.*
import java.io.IOException
import java.time.LocalDateTime
import java.time.format.DateTimeFormatter
import java.util.*

class MainActivity : AppCompatActivity(), NavigationView.OnNavigationItemSelectedListener {

    private val TAG = "myMessage"
    private var snackText: TextView? = null
    private var drinkText: TextView? = null
    private var priceText: TextView? = null
    private var dbHandler: MyDBHandler? = null

    @RequiresApi(Build.VERSION_CODES.O)
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        setSupportActionBar(toolbar)

        val randomizeButton = findViewById<Button>(R.id.randomizeButton)
        dbHandler = MyDBHandler(this, null, null, 1)
        val mjölkSwitch = findViewById<Switch>(R.id.mjölkSwitch)
        val nutSwitch = findViewById<Switch>(R.id.nutSwitch)
        val glutenSwitch = findViewById<Switch>(R.id.glutenSwitch)


        val toggle = ActionBarDrawerToggle(
                this, drawer_layout, toolbar, R.string.navigation_drawer_open, R.string.navigation_drawer_close)
        drawer_layout.addDrawerListener(toggle)
        toggle.syncState()

        nav_view.setNavigationItemSelectedListener(this)

        randomizeButton.setOnClickListener{

            try {

                dbHandler!!.createDataBase()

            } catch (ioe: IOException) {

                Log.i(TAG, "Unable to create database!")

            }

            val switchStateMjölk = mjölkSwitch.isChecked
            val switchStateNut = nutSwitch.isChecked
            val switchStateGluten = glutenSwitch.isChecked

            val list = dbHandler!!.getSnacks(switchStateMjölk, switchStateNut, switchStateGluten)
            val snacks = Arrays.copyOf<String, Any>(list.toTypedArray(), list.size, Array<String>::class.java)

            val list2 = dbHandler!!.getDrinks(switchStateMjölk, switchStateNut, switchStateGluten)
            val drinks = Arrays.copyOf<String, Any>(list2.toTypedArray(), list2.size, Array<String>::class.java)


            snackText = findViewById<TextView>(R.id.snackText)
            drinkText = findViewById<TextView>(R.id.drinkText)
            priceText = findViewById<TextView>(R.id.priceBox)
            val rand = Random()  //Set random class

            val theSnack = snacks[rand.nextInt(snacks.size)] //The final snack that you'll receive
            val theDrink = drinks[rand.nextInt(drinks.size)]//The final drink that you'll receive

            //while theSnack == lastSnack {
            //    val theSnack = snacks[rand.nextInt(snacks.size)]
            //}


            val handler = @SuppressLint("HandlerLeak")
            object : Handler() {
                override fun handleMessage(msg: Message) {
                    val randomNum = rand.nextInt(snacks.size)
                    val snack = snacks[randomNum]
                    snackText!!.text = snack

                    val snackPrice = dbHandler!!.getPriceSnack(snack)

                    val randomNum2 = rand.nextInt(drinks.size)
                    val drink = drinks[randomNum2]
                    drinkText!!.text = drink

                    val drinkPrice = dbHandler!!.getPriceDrink(drink)

                    val price = drinkPrice + snackPrice
                    val stringPrice = Integer.toString(price)
                    priceText!!.text = stringPrice + "kr"
                }
            }

            val handlerFinal = @SuppressLint("HandlerLeak")
            object : Handler() {
                override fun handleMessage(msg: Message) {
                    snackText!!.text = theSnack
                    drinkText!!.text = theDrink
                    Log.i(TAG, "Successful replacement of " + theSnack + " and " + theDrink)
                }
            }

            //test

            val r = Runnable {
                for (i in 0..49) {
                    handler.sendEmptyMessage(0)
                    try {
                        Thread.sleep(50)
                    } catch (ie: InterruptedException) {
                        Log.i(TAG, "Thread interrupted")
                    }
                }
                handlerFinal.sendEmptyMessage(0)
            }

            val myThread = Thread(r)
            myThread.start()

            val current = LocalDateTime.now()

            val formatter = DateTimeFormatter.ofPattern("yy-MM-dd")
            val formatted = current.format(formatter)
            val finalSnack = "'" + theSnack + "', "
            val finalDrink = "'" + theDrink + "', "
            val date = "'" + formatted + "'"
            dbHandler!!.addToHistorik(finalSnack, finalDrink, date)



        }
    }

    override fun onBackPressed() {
        if (drawer_layout.isDrawerOpen(GravityCompat.START)) {
            drawer_layout.closeDrawer(GravityCompat.START)
        } else {
            super.onBackPressed()
        }
    }

    override fun onCreateOptionsMenu(menu: Menu): Boolean {
        // Inflate the menu; this adds items to the action bar if it is present.
        menuInflater.inflate(R.menu.main, menu)
        return true
    }

    override fun onOptionsItemSelected(item: MenuItem): Boolean {
        // Handle action bar item clicks here. The action bar will
        // automatically handle clicks on the Home/Up button, so long
        // as you specify a parent activity in AndroidManifest.xml.
        when (item.itemId) {
            R.id.action_settings -> return true
            else -> return super.onOptionsItemSelected(item)
        }
    }

    override fun onNavigationItemSelected(item: MenuItem): Boolean {
        // Handle navigation view item clicks here.
        when (item.itemId) {
            R.id.nav_camera -> {
                val intent = Intent(this, HistoryActivity::class.java)
                startActivity(intent)
            }
            R.id.nav_gallery -> {

            }
            R.id.nav_slideshow -> {

            }
            R.id.nav_manage -> {

            }
            R.id.nav_share -> {

            }
            R.id.nav_send -> {

            }
        }

        drawer_layout.closeDrawer(GravityCompat.START)
        return true
    }

}
